odoo/cli/i18n.py

```diff
@@ -0,0 +1,164 @@
+        loadlang_parser.add_argument('-c', dest='config',
+            help='Use a specific configuration file')
+        loadlang_parser.add_argument(
+            '--languages', nargs='*', required=True,
+            help=('List of ISO codes to be installed.'))
+
+        # Parse args
+        try:
+            parsed_args, rest = self.parser.parse_known_args(args=cmdargs)
+        except ValueError as e:
+            self.parser.print_help()
+            Command.die(f'\n{e}\n')
+
+        # Read config
+        config_args = ["-c", parsed_args.config] if parsed_args.config else []
+        setup_logging = not ('to_stdout' in parsed_args and parsed_args.to_stdout)
+        config.parse_config(config_args, setup_logging=setup_logging)
+
+        db_names = odoo.tools.config['db_name']
+        if not db_names or len(db_names) > 1:
+            Command.die("Please provide a single database in the config file.")
+        db_name = db_names[0]
+
+        subcommand_handler = {
+            'import': self._import,
+            'export': self._export,
+            'loadlang': self._loadlang,
+        }[parsed_args.subcommand]
+
+        # Run subcommand
+        registry = Registry.new(db_name)
+        with closing(registry.cursor()) as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            subcommand_handler(env, parsed_args)
+
+    def _import(self, env, parsed_args):
+        if parsed_args.format in ('tgz', 'csv') and not parsed_args.lang:
+            Command.die("Language must be specified when '*.tgz' and '*.csv' files are imported.")
+        if parsed_args.format == 'csv':
+            # The default limit for CSV fields in the module is 128KiB, which is not
+            # quite sufficient to import images to store in attachment. 500MiB is a
+            # bit overkill, but better safe than sorry
+            csv.field_size_limit(500 << 20)
+
+        translation_importer = odoo.tools.translate.TranslationImporter(env.cr)
+        for filename in parsed_args.files:
+            if parsed_args.format == 'pot':
+                lang = parsed_args.lang
+            elif Path(filename).suffixes[-1:] == ['.po']:
+                code = Path(filename).stem
+                domain = ['|', ('code', '=', code), ('code', 'ilike', f'{code}%')]
+                lang = env['res.lang'].search(domain, limit=1).code
+            translation_importer.load_file(filename, lang)
+        translation_importer.save(overwrite=not parsed_args.only_new)
+
+    def _export(self, env, parsed_args, logging_enabled=True):
+
+        # When outputting to stdout, disable the logging
+        if parsed_args.to_stdout and logging_enabled:
+            logging.disable(logging.ERROR)
+            try:
+                self._export(env, parsed_args, logging_enabled=False)
+            finally:
+                logging.disable(logging.NOTSET)
+            return
+
+        if export_pot := 'pot' in parsed_args.lang:
```

Comment:
```suggestion
        export_pot = 'pot' in parsed_args.lang
        if export_pot:
```

When you use a walrus with a conditional, it hints that you're planning to only use the variable within the conditional code block, which is not the case here.

Reply:
We both know that python only scopes variables to the function, and not (like in C/JS) to the current "block" may that block be a function / an if / a loop. But as we are working with people with all backgrounds, it is better if we keep it closer to the *lingua-franca* (that is, the common understanding of what C does)